### PR TITLE
chore: List of avaliable online and offline stores exposed in test configuration

### PR DESCRIPTION
**Pull request by pyalex**
Signed-off-by: pyalex <moskalenko.alexey@gmail.com>

**What this PR does / why we need it**:

This will allow generating custom repo configs in specific test modules based on available online and offline stores.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/tests/integration/feature_repos/repo_configuration.py [hunk 1]
```diff
@@ -70,6 +73,21 @@
...
+    FileDataSourceCreator,
+]
+
+AVAILABLE_ONLINE_STORES: List[Any] = [
+    "sqlite",
+]
+
+if os.getenv("FEAST_IS_LOCAL_TEST", "False") != "True":
+    AVAILABLE_OFFLINE_STORES.extend(
+        [BigQueryDataSourceCreator, RedshiftDataSourceCreator]
```
Comment (felixwang9817) [inline sdk/python/tests/integration/feature_repos/repo_configuration.py:86]: missing snowflake

**Review by felixwang9817 — COMMENTED** [review]
I'm a bit confused by what this PR accomplishes - how does it make it easier to specify a set of configs for a test?

**Comment by pyalex** [comment]
@felixwang9817 
so usage could be like this
```
# test_logged_features.py

from tests.integration.feature_repos import AVAILABLE_OFFLINE_STORES

LOCAL_REPO_CONFIGS = [
   IntegrationTestRepoConfig(
        provider="gcp",
        offline_store_creator=BigQueryDataSourceCreator,
        online_store="redis",
        go_feature_retrieval=True,
    ),
   IntegrationTestRepoConfig(
        provider="aws",
        offline_store_creator=RedshiftDataSourceCreator,
        online_store=REDIS_CONFIG,
        go_feature_retrieval=True,
    ),

]

@pytest.fixture(params=[
    c for c in LOCAL_REPO_CONFIGS if c.offline_store_creator in AVAILABLE_OFFLINE_STORES
])
def local_environment():
     ...


def test_go_features_logging(local_environment):
    ...
```

**Comment by felixwang9817** [comment]
@pyalex got it, I think this is fine for now so I'll stamp. In the long run it still feels quite awkward to set up a `LOCAL_REPO_CONFIGS` in each test file (let alone per test); I like the decorators approach more.

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**